sda/internal/database/db_functions.go

```diff
@@ -62,6 +62,42 @@ func (dbs *SDAdb) getFileID(corrID string) (string, error) {
 	return fileID, nil
 }
 
+// GetInboxFilePathFromID checks if a file exists in the database for a given user and fileID
+// and that is not yet archived
+func (dbs *SDAdb) GetInboxFilePathFromID(submissionUser, fileID string) (string, error) {
+	var (
+		err      error
+		count    int
+		filePath string
+	)
+
+	for count == 0 || (err != nil && count < RetryTimes) {
+		filePath, err = dbs.getInboxFilePathFromID(submissionUser, fileID)
+		count++
+	}
+
+	return filePath, err
+}
+
+func (dbs *SDAdb) getInboxFilePathFromID(submissionUser, fileID string) (string, error) {
+	dbs.checkAndReconnectIfNeeded()
+	db := dbs.DB
+
+	const getFilePath = "SELECT submission_file_path from sda.files where " +
+		"submission_user= $1 and id = $2 " +
```

Comment:
This was briefly discussed when we started this task.While it's true that it's not needed, we decided to keep it as a double check of the file id. This could have been discussed more, but if we remove it here we might as well remove it from the endpoint.

Reply:
After the discussion on Slack, I think we should leave this for now. Feel free to create an issue if you would like it to be changed :+1: